var/spack/repos/builtin/packages/rocm-config/package.py

```diff
@@ -0,0 +1,175 @@
+    targets = []
+    try:
+        nodes = pathlib.Path("/sys/class/kfd/kfd/topology/nodes")
+        for filename in nodes.glob("*/properties"):
+            with filename.open(encoding="utf-8") as f:
+                for line in f:
+                    label = "gfx_target_version "
+                    if not line.startswith(label):
+                        continue
+                    version = int(line[len(label) :])
+                    if not version:
+                        break
+                    major_version = version // 10000
+                    minor_version = (version // 100) % 100
+                    step_version = version % 100
+                    target = "gfx{:d}{:x}{:x}".format(major_version, minor_version, step_version)
+                    if target in known_targets:
+                        targets.append(target)
+                    break
+    except (OSError, ValueError):
+        pass  # query is best-effort
+    if targets:
+        if multi:
+            return targets
+        else:
+            return targets[-1]
+    else:
+        return "none"
+
+def target_strip_feature(target):
+    return target.split(":")[0]
+
+class RocmConfig(BundlePackage):
+    """Target and Version configurations for ROCm Stacks"""
+
+    # ROCm stack versions available in Spack
+    rocm_versions = {
+        "5.4.3":  {"preferred": True},
+        "5.4.0":  {"preferred": False},
+        "5.3.3":  {"preferred": True},
+        "5.3.0":  {"preferred": False},
+        "5.2.2":  {"preferred": True},
+        "5.2.1":  {"preferred": False},
+        "5.2.0":  {"preferred": False},
+        "5.1.3":  {"preferred": True},
+        "5.1.0":  {"preferred": False},
+        # Deprecated Versions
+        "5.0.2":  {"deprecated": True},
+        "5.0.0":  {"deprecated": True},
+        "4.5.2":  {"deprecated": True},
+        "4.5.0":  {"deprecated": True},
+        "4.3.1":  {"deprecated": True},
+        "4.3.0":  {"deprecated": True},
+        "4.2.0":  {"deprecated": True},
+        "4.1.0":  {"deprecated": True},
+        "4.0.0":  {"deprecated": True},
+        "3.10.0": {"deprecated": True},
+        "3.9.0":  {"deprecated": True},
+        "3.8.0":  {"deprecated": True},
+        "3.7.0":  {"deprecated": True},
+        "3.5.0":  {"deprecated": True}
+    }
+
+    amdgpu_targets = (
+       # GCN GFX7 Sea Islands
+        "gfx700", "gfx701", "gfx702", "gfx703", "gfx704", "gfx705",
+        # GCN GFX8 Volcanic Islands
+        "gfx801", "gfx802", "gfx803", "gfx805", "gfx810",
+        # GCN GFX9 Vega
+        "gfx900", "gfx902", "gfx904", "gfx906", "gfx908", "gfx909", "gfx90a",
+        "gfx90c",
+        # GCN GFX10 RDNA 1
+        "gfx1010", "gfx1011", "gfx1012", "gfx1013",
+        # GCN GFX10 RDNA 2
+        "gfx1030", "gfx1031", "gfx1032", "gfx1033", "gfx1034", "gfx1035", "gfx1036",
+        # GCN GFX11 RDNA 3
+        "gfx1100", "gfx1101", "gfx1102", "gfx1103",
+    )
+
+    amdgpu_target_features = {
```

Comment:
I don't know if it is a good place, but I would like to know whether the hardware is an APU or a discrete GPU.

Related to https://github.com/kokkos/kokkos/pull/7462.